orttraining/orttraining/training_ops/cuda/nn/dropout.cc

```diff
@@ -87,6 +87,14 @@ Status DropoutGrad<T1, T2>::ComputeInternal(OpKernelContext* context) const {
   float ratio_data;
   auto ratio = context->Input<Tensor>(2);
 
+  auto training_mode = context->Input<Tensor>(3); // optional
+  bool training_mode_data = false;
+  if (training_mode){
+    std::cout << "CUDA DROPOUT BEFORE tensor->template Data<bool>() call" << std::endl;
+    ORT_ENFORCE(training_mode->Shape().Size() == 1);
+    training_mode_data = *(training_mode->template Data<bool>());
```

Comment:
@edgchen1 Any idea on why this line causes a segfault? orttraining/orttraining/training_ops/cpu/nn/dropout_op.cc line 105 has an equivalent line for CPU and it reads 'true' successfully.

Reply:
not really, are you able to get a core dump or run it with a debugger?

Reply:
is it possible because that the input data is on gpu?

Reply:
missing ` .InputMemoryType<OrtMemTypeCPUInput>(3)` for the new parameter was the cause!